### DISCUSSION TITLE
interacting with other users in the system

### DISCUSSION THREAD

**Discussion by folkertvanheusden** [adepierre/Botcraft#77] [Q&A]
Hi,

I would like to make it possible that other players send commands to my bot.
For that I need to know who send it.
See https://github.com/adepierre/Botcraft/issues/61
It looks like the GetFrom of a chat-message is always empty so I don't know who sent the command.
Also I don't think there's a way of finding who's logged in so that I could just iterate through the list and let people mention their name in the chat-message?
ClientboundChatPacket has a GetSender but I can't find a way of mapping a UUID-type to a username or Entity object?
I could invoke GetRawText() on the chat-message and interpret the json, but how can I find then the Entity-object that belongs to it? (to find the coordinates of the user).

**Comment by adepierre** [comment]
> I need to know who send it

Chat support is very limited, and only a very small part of the full [chat system](https://wiki.vg/Chat#Processing_chat) is implemented. I'd say your best move would indeed be to use GetRawText() on the chat object and directly parse the json from here to extract the name. Be careful though because all servers don't send chat in the same way (paper servers send a much more complicated json than vanilla for example). Your solution would thus not be compatible with any server. But if you know you'll always use it on one type of server, you should go with this approach.


> It looks like the GetFrom of a chat-message is always empty

You're right. I don't even remember what this from thing is or why it's here. That's one of those things that were added like 4 years ago and never changed since.


> I don't think there's a way of finding who's logged in

Correct. There is currently no support for the "tab" thing with names and ping of logged in players. I'm not even sure which packet is used for it.


> I could invoke GetRawText() on the chat-message and interpret the json, but how can I find then the Entity-object that belongs to it? (to find the coordinates of the user)

Basically you can't. At the moment, UUID are not stored in entities, correct. But even if they were, that wouldn't be a solution for you anyway because player entities only exist if you're "close" to them. Not for all logged in players. So if you receive a chat message from someone far away (i.e. not in the entity render distance range), you wouldn't have an associated entity. Neither in botcraft nor in vanilla minecraft.
